triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala

```diff
@@ -43,24 +55,61 @@ object TriggerServiceFixture {
       mat: Materializer,
       aesf: ExecutionSequencerFactory,
       ec: ExecutionContext): Future[A] = {
+    // Launch a toxiproxy instance . Wait on it to be ready to accept
+    // connections.
+    val toxiExe = BazelRunfiles.rlocation("external/toxiproxy_nix/bin/toxiproxy-cmd")
+    val toxiProxyPort = findFreePort()
+    val toxiCmd = toxiExe + " --port " ++ toxiProxyPort.toString
+    val toxiProxyProc = Process(toxiCmd).run()
+    var retries = 0
+    var scanning = true
+    while (scanning) {
+      try {
+        println("Waiting for Toxiproxy...")
+        val channel = new java.net.Socket("localhost", toxiProxyPort)
+        channel.close()
+        scanning = false
+      } catch {
+        case _: Throwable =>
+          retries = retries + 1
+          if (retries >= 3) {
+            println("Toxiproxy failed to start. Terminating with errors.")
+            System.exit(1) // Give up.
+          } else {
+            Thread.sleep(2000)
+          }
+      }
+    }
+    val toxiProxyClient = new ToxiproxyClient("localhost", toxiProxyPort);
 
     val ledgerId = LedgerId(testName)
     val applicationId = ApplicationId(testName)
     val ledgerF = for {
-      ledger <- Future(new SandboxServer(ledgerConfig(Port.Dynamic, dars, ledgerId), mat))
-      port <- ledger.portF
-    } yield (ledger, port.value)
+      ledger <- Future(new SandboxServer(ledgerConfig(Port(0), dars, ledgerId), mat))
+      sandboxPort <- ledger.portF
+      ledgerProxyPort <- Future(findFreePort())
+      ledgerProxy <- Future(
+        toxiProxyClient
+          .createProxy("sandbox", s"localhost:${ledgerProxyPort}", s"localhost:${sandboxPort}"))
+    } yield (ledger, ledgerProxyPort, ledgerProxy)
```

Comment:
The code below might be simplified if you omit the `ledger` and even `ledgerProxy`, which don't seem to be used.

Reply:
`ledger` is used when the future completes, `ledgerProxy` I anticipate using in the next set of changes.